### ISSUE TITLE
mixer/adapter/list TestRefreshAndPurge is flaky

### ISSUE THREAD

**Issue by hklai** [istio/istio#11020]
Example: https://gubernator.k8s.io/build/istio-prow/logs/istio-postsubmit/4176

```
env.go:117: Fetching list from http://127.0.0.1:41126
env.go:117: Installing updated list with 1 entries
env.go:117: Fetching list from http://127.0.0.1:41126
env.go:117: Fetched list is unchanged
env.go:117: Purging list due to inability to refresh it in time
list_test.go:533: Got success, expected error
```

Basically the same as https://github.com/istio/istio/issues/3686, which was never fixed.

**Comment by geeknoid** [comment]
PR pending.

**Comment by geeknoid** [comment]
Fixed by #11086
